core/src/test/java/io/parsingdata/metal/EqualityTest.java

```diff
@@ -196,24 +196,28 @@ public void parseGraph() {
 
     @Test
     public void stringRef() {
-        final Ref object = new NameRef("name");
-        assertFalse(object.equals(null));
+        final Ref<String> object = new NameRef("name");
+        assertNotEquals(null, object);
         assertNotEquals("name", object);
         assertEquals(object, new NameRef("name"));
+        assertNotEquals(object, new NameRef("name", "name"));
```

Comment:
I added this case, because I think that technically this should be equal. The result of these refs will always be the same, no matter how many times `"name"` was added as a reference. We could consider storing it in a Set first and then convert it to an ImmutableList.